### DISCUSSION TITLE
How hostname-patcher works?

### DISCUSSION THREAD

**Discussion by tamisbmxx4** [Bellisario/hostname-patcher#1] [Q&A] [answered]
hello sir i don't understand about this solution, i also have the same error

**Comment by Bellisario** [comment, marked as answer]
# Explaination
This is an error caused by Node.js retrieving "OS Hostname" (the name of the Operating System) whose function `os.hostname()` seems not to be implemented (broken) in Windows 7 devices using a not officially Node.js supported version (14 and above): the code just "implements" it by returning "localhost" every function call.
https://github.com/Bellisario/hostname-patcher/blob/38a01efd880752bdb7366b1a48246dc6df035964/lib/index.js#L11

# Why this works
Also if os module is required another time into another file this continue to work because modifying hostname function from the os module will make sure the new require will keep the patched function instead of the broken one.

# Should you be worried?
No, this is completely safe, because any change to os module will be lost after script runtime end.

# Does it break working Node.js machines?
No it doesn't, because this patch checks if `os.hostname()` is broken before patching, and only if it's broken, it will patch the function, leaving any other os module function as it is.
